packages/x-tree-view/src/hooks/useTreeItemModel.ts

```diff
@@ -0,0 +1,10 @@
+'use client';
+import { useTreeViewContext } from '../internals/TreeViewProvider';
+import { useSelector } from '../internals/hooks/useSelector';
+import { selectorItem } from '../internals/plugins/useTreeViewItems/useTreeViewItems.selectors';
+import { TreeViewItemId } from '../models';
+
+export const useTreeItemModel = <R extends {}>(itemId: TreeViewItemId) => {
+  const { store } = useTreeViewContext();
+  return useSelector(store, selectorItem, itemId) as R | null;
```

Comment:
If we don't make `useSelector` and `selectorItem` public for now, then I think it's important to provide a way to replace `publicAPI.getItem` calls in the render.